test/misc_ir.jl

```diff
@@ -227,31 +227,31 @@ x::T
 #---------------------
 LoweringError:
 ::T
-└─┘ ── `::` must be written `value::type` outside function argument lists
+└─┘ ── invalid syntax: unknown kind `::` or number of arguments (1)
```

Comment:
Many of these messages got better / clearer, but several seem to be somewhat worse now, in the sense that they either expose the user to more syntax- / lowering-isms (e.g. "bracescat" / "keyword-separating" / "kind") or refer to less specific constructs (e.g. "this syntax" vs. "\`module\`")

Do you think we can improve that?